drivers/sensor/tdk/icm4268x/icm4268x.h

```diff
@@ -0,0 +1,582 @@
+		{.val = 100000000, .reg = ICM42688_DT_GYRO_FS_1000},
+		{.val = 50000000,  .reg = ICM42688_DT_GYRO_FS_500},
+		{.val = 25000000,  .reg = ICM42688_DT_GYRO_FS_250},
+		{.val = 12500000,  .reg = ICM42688_DT_GYRO_FS_125},
+		{.val = 6250000,   .reg = ICM42688_DT_GYRO_FS_62_5},
+		{.val = 3125000,   .reg = ICM42688_DT_GYRO_FS_31_25},
+		{.val = 1562500,   .reg = ICM42688_DT_GYRO_FS_15_625},
+	},
+	[ICM4268X_VARIANT_ICM42686] = {
+		{.val = 400000000, .reg = ICM42686_DT_GYRO_FS_4000},
+		{.val = 200000000, .reg = ICM42686_DT_GYRO_FS_2000},
+		{.val = 100000000, .reg = ICM42686_DT_GYRO_FS_1000},
+		{.val = 50000000,  .reg = ICM42686_DT_GYRO_FS_500},
+		{.val = 25000000,  .reg = ICM42686_DT_GYRO_FS_250},
+		{.val = 12500000,  .reg = ICM42686_DT_GYRO_FS_125},
+		{.val = 6250000,   .reg = ICM42686_DT_GYRO_FS_62_5},
+		{.val = 3125000,   .reg = ICM42686_DT_GYRO_FS_31_25},
+	},
+};
+
+static inline uint8_t icm4268x_gyro_fs_to_reg(uint16_t dps, enum icm4268x_variant variant)
+{
+	for (uint8_t i = 0 ; i < table_gyro_fs_to_reg_array_size[variant] ; i++) {
+		if (dps * 100000 >= table_gyro_fs_to_reg[variant][i].val) {
+			return table_gyro_fs_to_reg[variant][i].reg;
+		}
+	}
+
+	/** Force values less than lower boundary */
+	return table_gyro_fs_to_reg[variant][table_gyro_fs_to_reg_array_size[variant] - 1].reg;
+}
+
+static inline void icm4268x_gyro_reg_to_fs(uint8_t fs, enum icm4268x_variant variant,
+					   struct sensor_value *out)
+{
+	for (uint8_t i = 0 ; i < table_gyro_fs_to_reg_array_size[variant] ; i++) {
+		if (fs == table_gyro_fs_to_reg[variant][i].reg) {
+			sensor_10udegrees_to_rad(table_gyro_fs_to_reg[variant][i].val, out);
+			return;
+		}
+	}
+
+	CODE_UNREACHABLE;
+}
+
+static inline uint8_t icm4268x_accel_hz_to_reg(uint16_t hz)
+{
+	if (hz >= 32000) {
+		return ICM4268X_DT_ACCEL_ODR_32000;
+	} else if (hz >= 16000) {
+		return ICM4268X_DT_ACCEL_ODR_16000;
+	} else if (hz >= 8000) {
+		return ICM4268X_DT_ACCEL_ODR_8000;
+	} else if (hz >= 4000) {
+		return ICM4268X_DT_ACCEL_ODR_4000;
+	} else if (hz >= 2000) {
+		return ICM4268X_DT_ACCEL_ODR_2000;
+	} else if (hz >= 1000) {
+		return ICM4268X_DT_ACCEL_ODR_1000;
+	} else if (hz >= 500) {
+		return ICM4268X_DT_ACCEL_ODR_500;
+	} else if (hz >= 200) {
+		return ICM4268X_DT_ACCEL_ODR_200;
+	} else if (hz >= 100) {
+		return ICM4268X_DT_ACCEL_ODR_100;
+	} else if (hz >= 50) {
+		return ICM4268X_DT_ACCEL_ODR_50;
+	} else if (hz >= 25) {
+		return ICM4268X_DT_ACCEL_ODR_25;
+	} else if (hz >= 12) {
+		return ICM4268X_DT_ACCEL_ODR_12_5;
+	} else if (hz >= 6) {
+		return ICM4268X_DT_ACCEL_ODR_6_25;
+	} else if (hz >= 3) {
+		return ICM4268X_DT_ACCEL_ODR_3_125;
+	} else {
+		return ICM4268X_DT_ACCEL_ODR_1_5625;
+	}
+}
+
+static inline void icm4268x_accel_reg_to_hz(uint8_t odr, struct sensor_value *out)
+{
+	switch (odr) {
+	case ICM4268X_DT_ACCEL_ODR_32000:
+		out->val1 = 32000;
+		out->val2 = 0;
+		return;
+	case ICM4268X_DT_ACCEL_ODR_16000:
+		out->val1 = 1600;
```

Comment:
```suggestion
		out->val1 = 16000;
```

Reply:
Thanks for spotting this. Addressed in #93742